ykcompile/src/lib.rs

```diff
@@ -633,6 +633,86 @@ impl<TT> TraceCompiler<TT> {
         Location::Register(dst)
     }
 
+    fn c_cast(&mut self, op: &Operand, cty: &Ty) -> Location {
+        match op {
+            Operand::Place(p) => {
+                let (loc, ty) = self.place_to_location(p, false);
+                let temp = self.create_temporary();
+                match loc {
+                    Location::Register(reg) => {
+                        match cty.size() {
+                            1 => {
+                                dynasm!(self.asm
+                                    ; mov Rb(temp), Rb(reg)
+                                );
+                            }
+                            2 => match ty.size() {
+                                1 => {
+                                    dynasm!(self.asm
+                                        ; movzx Rw(temp), Rb(reg)
+                                    );
+                                }
+                                2 | 4 | 8 => {
+                                    dynasm!(self.asm
+                                        ; mov Rw(temp), Rw(reg)
+                                    );
+                                }
+                                _ => todo!("{}", ty.size()),
+                            },
+                            4 => match ty.size() {
+                                1 => {
+                                    dynasm!(self.asm
+                                        ; movzx Rd(temp), Rb(reg)
+                                    );
+                                }
+                                2 => {
+                                    dynasm!(self.asm
+                                        ; movzx Rd(temp), Rw(reg)
+                                    );
+                                }
+                                4 | 8 => {
+                                    dynasm!(self.asm
+                                        ; mov Rd(temp), Rd(reg)
+                                    );
+                                }
+                                _ => todo!("{}", ty.size()),
+                            },
+                            8 => {
+                                match ty.size() {
+                                    1 => {
+                                        dynasm!(self.asm
+                                            ; movzx Rq(temp), Rb(reg)
```

Comment:
What about signed integers?

Reply:
Then the value should be sign extended. Good catch.

Reply:
I've split `c_cast` into multiple functions and added a `todo!` for now. Commit coming soon once I figured out what's going on with ykrustc.

Reply:
"Fixed" in 19fe606